### ISSUE TITLE
3.3.0 attempt to index local 'category' (a nil value)

### ISSUE THREAD

**Issue by danez** [BreakBB/ExtendedCharacterStats#309]
## Bug description

Lua error on start in classic era 1.15.1

I think this is caused by the new #305 feature

Maybe because my character is hunter? not sure

Also after the error the stats are pretty much empty besides meele AP, hit bonus and crit chance.

## Screenshots

<img width="2429" alt="Screenshot 2024-03-26 at 18 50 46" src="https://github.com/BreakBB/ExtendedCharacterStats/assets/231804/a90be3d2-bba0-4431-a5af-acf12c695320">

```
Message: ...face/AddOns/ExtendedCharacterStats/Modules/Stats.lua:212: attempt to index local 'category' (a nil value)
Time: Tue Mar 26 18:46:29 2024
Count: 1
Stack: ...face/AddOns/ExtendedCharacterStats/Modules/Stats.lua:212: attempt to index local 'category' (a nil value)
[string "@Interface/AddOns/ExtendedCharacterStats/Modules/Stats.lua"]:212: in function <...face/AddOns/ExtendedCharacterStats/Modules/Stats.lua:193>
[string "@Interface/AddOns/ExtendedCharacterStats/Modules/Stats.lua"]:113: in function `CreateWindow'
[string "@Interface/AddOns/ExtendedCharacterStats/Modules/Init.lua"]:31: in function `OnPlayerLogin'
[string "@Interface/AddOns/ExtendedCharacterStats/ECS.lua"]:36: in function <Interface/AddOns/ExtendedCharacterStats/ECS.lua:29>

Locals: profile = <table> {
 ranged = <table> {
 }
 defense = <table> {
 }
 general = <table> {
 }
 spell = <table> {
 }
 spellBonus = <table> {
 }
 regen = <table> {
 }
 melee = <table> {
 }
}
category = nil
(*temporary) = <function> defined @Interface/AddOns/ExtendedCharacterStats/Modules/Stats.lua:161
(*temporary) = nil
(*temporary) = <table> {
 text = "Rating"
 statColor = "558b2f"
 display = true
 isTbcOnly = true
 refName = "MeleeHitRating"
 textColor = "85bb5c"
}
(*temporary) = <table> {
 statColor = "558b2f"
 display = true
 text = "Bonus"
 refName = "MeleeHitBonus"
 textColor = "85bb5c"
}
(*temporary) = <table> {
 statColor = "558b2f"
 display = false
 text = "Miss"
 refName = "MeleeHitSameLevel"
 textColor = "85bb5c"
}
(*temporary) = <table> {
 statColor = "558b2f"
 display = false
 text = "Miss (Lvl + 3)"
 refName = "MeleeHitBossLevel"
 textColor = "85bb5c"
}
(*temporary) = <table> {
 bossLevel = <table> {
 }
 isSubGroup = true
 display = true
 rating = <table> {
 }
 sameLevel = <table> {
 }
 text = "Hit"
 refName = "MeleeHitHeader"
 bonus = <table> {
 }
}
(*temporary) = <table> {
 1 = <table> {
 }
 2 = <table> {
 }
 3 = <table> {
 }
 4 = <table> {
 }
}
(*temporary) = <function> defined =[C]:-1
(*temporary) = <table> {
 1 = <table> {
 }
 2 = <table> {
 }
 3 = <table> {
 }
 4 = <table> {
 }
}
(*temporary) = 4
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = <table> {
 statColor = "558b2f"
 display = false
 text = "Miss (Lvl + 3)"
 refName = "MeleeHitBossLevel"
 textColor = "85bb5c"
}
(*temporary) = "table"
(*temporary) = true
(*temporary) = "attempt to index local 'category' (a nil value)"
_CreateStatInfo = <function> defined @Interface/AddOns/ExtendedCharacterStats/Modules/Stats.lua:161

```
## Version

3.3.0

**Comment by danez** [comment]
I tried increasing the [profile version](https://github.com/BreakBB/ExtendedCharacterStats/blob/master/Modules/Profile.lua#L8), but that did not help.

**Comment by danez** [comment]
Created a Pr for the fix #310

**Comment by Ooshga** [comment]
Same for me. Happened after the update.


Message: ...Ons/ExtendedCharacterStats/Modules/Config/Config.lua:244: attempt to index global 'ECSConfigFrame' (a nil value)
Time: Tue Mar 26 19:24:51 2024
Count: 1
Stack: ...Ons/ExtendedCharacterStats/Modules/Config/Config.lua:244: attempt to index global 'ECSConfigFrame' (a nil value)
[string "@Interface/AddOns/ExtendedCharacterStats/Modules/Config/Config.lua"]:244: in function `ToggleWindow'
[string "@Interface/AddOns/ExtendedCharacterStats/Modules/SlashCommands.lua"]:20: in function `?'
[string "@Interface/FrameXML/ChatFrame.lua"]:4975: in function <Interface/FrameXML/ChatFrame.lua:4921>
[string "=[C]"]: in function `ChatEdit_ParseText'
[string "@Interface/FrameXML/ChatFrame.lua"]:4638: in function <Interface/FrameXML/ChatFrame.lua:4637>
[string "=[C]"]: in function `ChatEdit_SendText'
[string "@Interface/FrameXML/ChatFrame.lua"]:4674: in function `ChatEdit_OnEnterPressed'
[string "*ChatFrame.xml:121_OnEnterPressed"]:1: in function <[string "*ChatFrame.xml:121_OnEnterPressed"]:1>

Locals: self = <table> {
 private = <table> {
 }
 CreateWindow = <function> defined @Interface/AddOns/ExtendedCharacterStats/Modules/Config/Config.lua:27
 ToggleWindow = <function> defined @Interface/AddOns/ExtendedCharacterStats/Modules/Config/Config.lua:243
}
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index global 'ECSConfigFrame' (a nil value)"
